### PR TITLE
Add Availability Zones for Openstack

### LINKED ISSUE #2390: UI/API for Multi AZ support for OpenStack
**User Story**
OpenStack datacenters contain currently a fixed availability zone in the datacenters.yaml / CRD. To support HA clusters over multiple AZs, the zone should be selectable when creating a machine deployment.

This is already supported by Kubermatic. Editing a machine-deployment from the UI currently sets it to the AZ, configured for the datacenter.

**Acceptance criteria**
Availability zone is selectable during machine deployment creation from the UI

### PR DESCRIPTION
**Pull request by kgroschoff**
**What this PR does / why we need it**:
Add Availability Zones for Openstack

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2390

**Special notes for your reviewer**:
https://github.com/kubermatic/kubermatic/pull/5535 needs to get merged first
/hold 

**Release note**:

```release-note
Added possibility to specify Availability Zones for Openstack via UI
```

### CODE REVIEW

File: src/app/cluster/cluster-details/node-list/node-list.component.html [hunk 1]
```diff
@@ -228,6 +228,11 @@
               <div value>{{element.spec.cloud.openstack.image}}</div>
             </km-property>
 
+            <km-property *ngIf="element.spec.cloud.openstack && element.spec.cloud.openstack.availabilityZone">
```
Comment (maciaszczykm) [inline src/app/cluster/cluster-details/node-list/node-list.component.html:231]: In the HTML templates you can use `?.` for null checks, i.e. `element.spec.cloud.openstack?.availabilityZone` should do the same as above. 

https://angular.io/guide/template-syntax#safe-navigation-operator

**Comment by maciaszczykm** [comment]
LGTM, I will leave it to @floreks as he is already working with node data.

**Comment by kdomanski** [comment]
@floreks bitte review wkrótce.

**Comment by floreks** [comment]
Once i fix e2e so we can actually merge anything :)

**Pull request merged**